gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/UserTaskControllerTest.java

```diff
@@ -334,4 +336,386 @@ public void shouldYieldInternalErrorWhenRejectionInternal(final RejectionType re
+        .isEqualTo(expectedBody);
+
+    final var argumentCaptor = ArgumentCaptor.forClass(BrokerUserTaskAssignmentRequest.class);
+    Mockito.verify(brokerClient).sendRequest(argumentCaptor.capture());
+    Assertions.assertThat(argumentCaptor.getValue().getRequestWriter())
+        .hasUserTaskKey(1L)
+        .hasAction("assign")
+        .hasAssignee("Test Assignee");
+  }
+
+  @ParameterizedTest
+  @EnumSource(
+      value = RejectionType.class,
+      names = {"PROCESSING_ERROR", "EXCEEDED_BATCH_RECORD_SIZE", "SBE_UNKNOWN", "NULL_VAL"})
+  public void shouldYieldInternalErrorWhenInternalRejectionForTaskAssignment(
+      final RejectionType rejectionType) {
+    // given
+    brokerResponseFutureSupplier =
+        () ->
+            CompletableFuture.supplyAsync(
+                () ->
+                    new BrokerRejectionResponse<>(
+                        new BrokerRejection(
+                            UserTaskIntent.ASSIGN, 1L, rejectionType, "Just an error")));
+
+    final var request = new UserTaskAssignmentRequest().assignee("Test Assignee");
+    final var expectedBody =
+        ProblemDetail.forStatusAndDetail(
+            HttpStatus.INTERNAL_SERVER_ERROR,
+            "Command 'ASSIGN' rejected with code '" + rejectionType + "': Just an error");
+    expectedBody.setTitle(rejectionType.name());
+    expectedBody.setInstance(URI.create("/api/v1/user-tasks/1/assignment"));
+
+    // when / then
+    webClient
+        .post()
+        .uri("api/v1/user-tasks/1/assignment")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(Mono.just(request), UserTaskAssignmentRequest.class)
+        .exchange()
+        .expectStatus()
+        .is5xxServerError()
+        .expectBody(ProblemDetail.class)
+        .isEqualTo(expectedBody);
+
+    final var argumentCaptor = ArgumentCaptor.forClass(BrokerUserTaskAssignmentRequest.class);
+    Mockito.verify(brokerClient).sendRequest(argumentCaptor.capture());
+    Assertions.assertThat(argumentCaptor.getValue().getRequestWriter())
+        .hasUserTaskKey(1L)
+        .hasAction("assign")
+        .hasAssignee("Test Assignee");
+  }
+
+  @Test
+  public void shouldYieldBadRequestWhenNoAssigneeForTaskAssignment() {
+    // given
+    brokerResponseFutureSupplier =
+        () ->
+            CompletableFuture.supplyAsync(
+                () ->
+                    new BrokerRejectionResponse<>(
+                        new BrokerRejection(
+                            UserTaskIntent.ASSIGN,
+                            1L,
+                            RejectionType.INVALID_ARGUMENT,
+                            "No assignee provided")));
+
+    final var request = new UserTaskAssignmentRequest();
+
+    final var expectedBody =
+        ProblemDetail.forStatusAndDetail(
+            HttpStatus.BAD_REQUEST,
+            "Command 'ASSIGN' validation failed with code 'INVALID_ARGUMENT': No assignee provided");
```

Comment:
🔧 After the simplification suggestion, this will be simpler as well.
```suggestion
            "No assignee provided");
```